sql/core/src/test/resources/sql-tests/results/pipe-operators.sql.out

```diff
@@ -678,6 +678,156 @@ org.apache.spark.sql.AnalysisException
 }
 
 
+-- !query
+table t
+|> set x = 1
+-- !query schema
+struct<y:string,x:int>
```

Comment:
so the new columns in SET are always put at the end instead of in-place update?

Reply:
Good catch. I was using "SELECT * EXCEPT" and appending the new expression at the end to implement the behavior. It turns out this changed the column order, so I updated the `UnresolvedStarExcept` expression to support performing the replacement in the original column position instead. This expression is now named `UnresolvedStarExceptOrReplace`.

Note that outside of SQL pipe operators, we could choose to use this to support "SELECT * EXCEPT REPLACE" syntax [1] in the future if we want. I am not implementing that now in the PR in the interest of keeping the PRs small and focused, but leaving a note here that we could do it in the future if desired.

[1] https://stackoverflow.com/questions/73514343/databases-that-support-select-except-replace